server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/FileAttributes.java

```diff
@@ -31,10 +33,13 @@
   private final boolean unitTest;
   @CheckForNull
   private final String languageKey;
+  private final int lines;
 
-  public FileAttributes(boolean unitTest, @Nullable String languageKey) {
+  public FileAttributes(boolean unitTest, @Nullable String languageKey, int lines) {
     this.unitTest = unitTest;
     this.languageKey = languageKey;
+    checkArgument(lines > 0, "Lines has not been set for this file");
```

Comment:
The text of the error makes assumption of where the constructor is called, which is too much.

See my comment on `getLines()`, you're adding this check to ensure comply with the contract this class advertises, the message should be related to the contract. The error should be "number of line must be >= 0".

If you want to throw this error message (I think it's a good idea), put it where applicable, ie. in ComponentRootBuilder.

BTW, I'm not sure a file can't be empty. See my other comment.